src/ecs.ts

```diff
@@ -0,0 +1,59 @@
+import { ComponentIter } from './ecs';
+import { transpose } from 'ramda';
+
+
+export class Entity {
+  components: Record<string, number> = {};
+}
+
+export class Component {}
+export type ComponentIter<T> = T[];
+
+export interface System <T extends Component[]> {
+  components: Component[];
+  onUpdate(componentIter: ComponentIter<T>);
+}
+
+export class ECS {
+  public entities: Entity[] = [];
+  public components: Record<string, Component[]> = {};
+  public systems: System<any>[] = [];
+
+  public createEntity(): Entity {
+    const entity = new Entity();
+    this.entities.push(entity);
+    return entity;
+  }
+
+  public registerComponent(component: Component): ECS {
+    this.components[component.constructor.name] = [];
+    return this;
+  }
+
+  public registerSystem <T extends Component[]> (system: System<T>): ECS {
```

Comment:
Presumably, `<T extends Component[]>` is supposed to prevent classes that do not extend Component? I don't think it works though. I was able to this: 
``` javascript 
class Health { // does not extend Component
  value: number;
  constructor(value: number = 100) {
    this.value = value
  }
}

class VelocitySystem implements System<[Health, Velocity]> {
  components = [new Health, new Velocity];
}
```

See https://github.com/ryanhinkel/pixistorm/blob/non-components-can-be-put-in-array/src/ecs.test.ts#L27-L37 for more.